src/main/java/org/broadinstitute/consent/http/models/Dataset.java

```diff
@@ -290,22 +303,51 @@ public void setDeletable(Boolean deletable) {
     this.deletable = deletable;
   }
 
+  // helper methods for open access string matching
+  public Optional<DatasetProperty> getPropertyByName(String name) {
```

Comment:
The only use of this is in the string matching method - we should move the logic there.